src/UniversalRewardsDistributor.sol

```diff
@@ -88,54 +86,48 @@ contract UniversalRewardsDistributor is IUniversalRewardsDistributor {
         }
     }
 
-    /// @notice Accepts and sets the current pending merkle root.
-    /// @dev This function can only be called after the timelock has expired.
+    /// @notice Accepts and sets the pending Merkle root.
+    /// @dev This function can only be called after the timelock has elapsed.
     /// @dev Anyone can call this function.
     function acceptRoot() external {
         require(pendingRoot.submittedAt > 0, ErrorsLib.NO_PENDING_ROOT);
-        require(block.timestamp >= pendingRoot.submittedAt + timelock, ErrorsLib.TIMELOCK_NOT_EXPIRED);
+        require(block.timestamp >= pendingRoot.submittedAt + timelock, ErrorsLib.TIMELOCK_NOT_ELAPSED);
 
-        root = pendingRoot.root;
-        ipfsHash = pendingRoot.ipfsHash;
-
-        emit EventsLib.RootSet(pendingRoot.root, pendingRoot.ipfsHash);
-
-        delete pendingRoot;
+        _setRoot(pendingRoot.root, pendingRoot.ipfsHash);
     }
 
     /// @notice Claims rewards.
     /// @param account The address to claim rewards for.
     /// @param reward The address of the reward token.
-    /// @param claimable The overall claimable amount of token rewards.
-    /// @param proof The merkle proof that validates this claim.
-    /// @return amount The amount of reward token claimed.
+    /// @param totalEarned The total amount of token rewards earned.
+    /// @param proof The Merkle proof that validates this claim.
+    /// @return claimedAmount The amount of reward token claimed.
     /// @dev Anyone can claim rewards on behalf of an account.
-    function claim(address account, address reward, uint256 claimable, bytes32[] calldata proof)
+    function claim(address account, address reward, uint256 totalEarned, bytes32[] calldata proof)
         external
-        returns (uint256 amount)
+        returns (uint256 claimedAmount)
     {
-        require(root != bytes32(0), ErrorsLib.ROOT_NOT_SET);
         require(
             MerkleProof.verifyCalldata(
-                proof, root, keccak256(bytes.concat(keccak256(abi.encode(account, reward, claimable))))
+                proof, root, keccak256(abi.encode(keccak256(abi.encode(account, reward, totalEarned))))
             ),
-            ErrorsLib.INVALID_PROOF_OR_EXPIRED
+            ErrorsLib.INVALID_PROOF
```

Comment:
The proof is invalid for this tree, and we don't have a notion of expiration built-in (it could be rebuilt on top by pruning the tree but nothing tells us that's what the updaters will decide to do)